### PR TITLE
Bump OpenTelemetry core packages to 1.3.0 and non-core to 1.0.0-rc9.4

### LINKED ISSUE #684: Update OpenTelemetry to 1.3.0
Update OpenTelemetry to 1.3.0 when it will be ready.

This release is expected before end of May and brings official support for Metrics.

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #684

Changes proposed in this pull request:

Bump OpenTelemetry core packages to 1.3.0 and non-core to 1.0.0-rc9.4

### CODE REVIEW

**Comment by pellared** [comment]
It may be good to update `CHANGELOG.md`.

File: test/test-applications/integrations/TestApplication.MongoDB/TestApplication.MongoDB.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="6.0.0" />
 
     <!-- Hack to fix SDK dependencies -->
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.0.3" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
```
[the remark this answers is not in the document]
Comment (Kielek) [reply test/test-applications/integrations/TestApplication.MongoDB/TestApplication.MongoDB.csproj:22]: Separate issue created: #719

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
Have you checked if all the [examples](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/tree/main/examples) are still working?

**Review by nrcventura — APPROVED** [review]

**Comment by Kielek** [comment]
> Have you checked if all the [examples](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/tree/main/examples) are still working?

Most of them. There is one issue, visible also on current main. Tracked under #820

**Pull request merged**